src/main/java/com/ait/lienzo/shared/core/types/Color.java

```diff
@@ -259,27 +268,7 @@ public static Color fromColorString(String cssColorString)
         {
             if (str.startsWith("#"))
             {
-                String r, g, b;
-
-                if (str.length() == 7)
-                {
-                    r = str.substring(1, 3);
-                    g = str.substring(3, 5);
-                    b = str.substring(5, 7);
-                }
-                else if (str.length() == 4)
-                {
-                    r = str.substring(1, 2);
-                    g = str.substring(2, 3);
-                    b = str.substring(3, 4);
-
-                    r = r + r;
-                    g = g + g;
-                    b = b + b;
-                }
-                else return null;// error - invalid length
-
-                return new Color(fixRGB(Integer.valueOf(r, 16)), fixRGB(Integer.valueOf(g, 16)), fixRGB(Integer.valueOf(b, 16)));
+                return hex2RGB(cssColorString);
```

Comment:
The TODO in hex2RGB method was about this functionality, so I just moved it to appropriate place. So we have two benefits:
1) Reduced code duplication
2) implemented TODO

Both methods have tests now. So it have to be safety change.